lib/kramdown/converter/math_engine/sskatex/details.rb

```diff
@@ -0,0 +1,264 @@
+#
+module ::Kramdown::Converter::MathEngine::SsKaTeX
+  class << self   # defining class methods
+    private
+
+    # Used for the default configuration
+    DATADIR = File.expand_path('../../../../../data/kramdown/sskatex',
+                               File.dirname(__FILE__))
+
+    # The default for the :jsdir option
+    DEFAULT_JSDIR = File.join(DATADIR, 'js')
+
+    # The default for the :katexjs option
+    DEFAULT_KATEXJS = File.join('katex', 'katex.min.js')
+
+    # The default for the :libs option
+    DEFAULT_LIBS =
+      ['object_assign.js', 'escape_nonascii_html.js', 'tex_to_html.js']
+
+    # User-overridable KaTeX options.
+    # Note that +displayMode+ is computed dynamically.
+    # Typical keys: +throwOnError+, +errorColor+, +colorIsTextColor+, +macros+.
+    DEFAULT_KATEXOPTS = {}
+
+    # Class-level cache for JS engine context, queried by configuration.
+    # Note: JSCTX contents may become stale if the contents of used JS files
+    # change and the configuration remains unchanged.
+    JSCTX = {}
+
+    # ExecJS uses <tt>runtime = Runtimes.const_get(name)</tt> without checks.
+    # That is fragile and potentially insecure with arbitrary user input.
+    # Instead we use a fixed dictionary restricted to valid contents.
+    # Note that there are aliases like <tt>SpiderMonkey = Spidermonkey</tt>.
+    JSRUN_FROMSYM =
+      ExecJS::Runtimes.constants.inject({}) { |dict, name|
+        runtime = ExecJS::Runtimes.const_get(name)
+        dict[name] = runtime if runtime.is_a?(ExecJS::Runtime)
+        dict }
+
+    # This should really be provided by +ExecJS::Runtimes+:
+    # A map from JS Runtime classes to their respective symbols.
+    # (+.name+ is too verbose.)
+    # For aliases like <tt>SpiderMonkey = Spidermonkey</tt>,
+    # there are several possible choices for the symbol,
+    # but all are equivalent, so we do not care.
+    JSRUN_TOSYM = JSRUN_FROMSYM.invert
+
+    # Dictionary for escape sequences used in Javascript string literals
+    JS_ESCAPE = {
+      "\\" => "\\\\",
+      "\"" => "\\\"",
+      # Escaping single quotes not necessary in double-quoted string literals
+      #"'" => "\\'",
+      # JS does not recognize \a nor GNU's \e
+      # \b is ambiguous in regexps, as in Perl and Ruby
+      #"\b" => "\\b",
+      "\f" => "\\f",
+      "\n" => "\\n",
+      "\r" => "\\r",
+      "\t" => "\\t",
+      "\v" => "\\v",
+    }
+
+    # Turn a string into an equivalent Javascript literal, double-quoted.
+    # Similar to +.to_json+, but escape all non-ASCII codes as well.
+    def jsquote(str)
+      # No portable way of escaping Unicode points above 0xFFFF
+      '"%s"' % str.gsub(/[\0-\u{001F}""\\\u{0080}-\u{FFFF}]/u) { |c|
+        JS_ESCAPE[c] || "\\u%04x" % c.ord
+      }
+    end
+
+    # This should really be provided by +ExecJS::Runtimes+:
+    # A list of available JS engines, as symbols, in the order of preference.
+    def jsruns_available
```

Comment:
Shouldn't this be called `available_jsruns` since it returns the available JS runtimes?